text/0000-default-field-values.md

```diff
@@ -0,0 +1,1658 @@
+  pattern syntax. As aforementioned, destruction (and pattern matching)
+  generally attempts to follow construction in Rust. Because of that,
+  introducing `Foo { .. }` is essentially cost-free in terms of the complexity
+  budget. It is arguably even cost-negative.
+
++ By writing `Foo { .. }`, you have there is explicit indication that default
+  values are being used; this enhances local reasoning further.
+
+# Prior art
+[prior-art]: #prior-art
+
+## Other languages
+
+This selection of languages are not exhaustive; rather, a few notable or
+canonical examples are used instead.
+
+### Java
+
+In Java it is possible to assign default values, computed by any expression,
+to an instance variable; for example, you may write:
+
+```java
+class Main {
+    public static void main(String[] args) {
+        new Foo();
+    }
+
+    public static int make_int() {
+        System.out.println("I am making an int!");
+        return 42;
+    }
+
+    static class Foo {
+        private int bar = Main.make_int();
+    }
+}
+```
+
+When executing this program, the JVM will print the following to `stdout`:
+
+```
+I am making an int!
+```
+
+Two things are worth noting here:
+
+1. It is possible to cause arbitrary side effects in the expression that
+   computes the default value of `bar`. This behaviour is unlike that which
+   this RFC proposes.
+
+2. It is possible to construct a `Foo` which uses the default value of `bar`
+   even though `bar` has `private` visibility. This is because default values
+   act as syntactic sugar for how the default constructor `Foo()` should act.
+   There is no such thing as constructors in Rust. However, the behaviour
+   that Java has is morally equivalent to this RFC since literals are
+   constructor-like and because this RFC also permits the usage of defaults
+   for private fields where the fields are not visible.
+
+### Scala
+
+Being a JVM language, Scala builds upon Java and retains the notion of default
+field values. For example, you may write:
+
+```scala
+case class Person(name: String = make_string(), age: Int = 42)
+
+def make_string(): String = {
+    System.out.println("foo");
+    "bar"
+}
+
+var p = new Person(age = 24);
+System.out.println(p.name);
+```
+
+As expected, this prints `foo` and then `bar` to the terminal.
+
+### Kotlin
+
+Kotlin is similar to both Java and Scala; here too can you use defaults:
+
+```kotlin
+fun make_int(): Int {
+    println("foo");
+    return 42;
+}
+
+class Person(val age: Int = make_int());
+
+fun main() {
+    Person();
+}
+```
+
+Similar to Java and Scala, Kotlin does also permit side-effects in the default
+values because both languages have no means of preventing the effects.
+
+### C#
+
+Another language with defaults in the OO list is C#.
+The behaviour similar to Java:
+
+```csharp
+class Foo {
+    int bar = 42;
+}
+```
+
+### C++
+
+Another language in the object oriented family is C++. It also affords default
```

Comment:
`object-oriented`. Maybe write out the abbreviation `OO` above too, just in case?